css/amp.css

```diff
@@ -422,6 +422,16 @@ i-amphtml-sizer {
 .i-amphtml-element > [placeholder].amp-hidden {
   visibility: hidden;
 }
+/* Placeholder, by design, expands layout=container unlike other layouts where
+   it is position:absolute and overlays the layout box.
+   However, when it goes away it should no longer expand the parent, so instead
+   of visibility:hidden it must be display:none */
+.i-amphtml-layout-container > [placeholder].hidden,
+[layout=container] > [placeholder].hidden,
```

Comment:
Minor: Are the `[layout=container] > ...` rules necessary? 

I think that it targets not-built elements, but not-built elements should display the placeholder. So I think `[layout=container] > [placeholder].hidden` may not actually affect any elements in practice.

E.g. existing rules that hide the placeholder also don't use `[layout]` attribute selector:
https://github.com/ampproject/amphtml/blob/cd1d9380ce4a73250ef6f21111bbe4b8b84fdf72/css/amp.css#L421-L424

Reply:
yeah you are right. `amp-list` when it becomes container does not seem to append `.i-amphtml-layout-container`. regardless I will hold on merging this until a decision is made on https://github.com/ampproject/amphtml/issues/18849

Reply:
agreed and done.